insights/parsers/smt.py

```diff
@@ -0,0 +1,117 @@
+"""
+Simultaneous Multithreading (SMT) parsers
+=========================================
+
+Parsers included in this module are:
+
+CpuSMTActive - file ``/sys/devices/system/cpu/smt/active``
+----------------------------------------------------------
+CpuCoreOnline - files matching ``/sys/devices/system/cpu/cpu[0-9]*/online``
+---------------------------------------------------------------------------
+CpuSiblings - files matching ``/sys/devices/system/cpu/cpu[0-9]*/topology/thread_siblings_list``
+------------------------------------------------------------------------------------------------
+"""
+
+from insights.core import Parser
+from insights.core.plugins import parser
+from insights.parsers import SkipException
+from insights.specs import Specs
+import re
+
+
+@parser(Specs.cpu_smt_active)
+class CpuSMTActive(Parser):
+    """
+    Class for parsing ``/sys/devices/system/cpu/smt/active`` file.
+    Reports whether SMT is enabled and active.
+
+    Typical output of this command is::
+
+        1
+
+    Raises:
+        SkipException: When content is empty or cannot be parsed.
+
+    Examples:
+        >>> cpu_smt.on
+        True
+    """
+    def parse_content(self, content):
+        if not content:
+            raise SkipException("No content.")
+        self.on = bool(int(content[0]))
+
+
+@parser(Specs.cpu_cores)
+class CpuCoreOnline(Parser):
+    """
+    Class for parsing ``/sys/devices/system/cpu/cpu[0-9]*/online`` matching files.
+    Reports whether a CPU core is online.
+
+    Typical output of this command is::
+
+        1
```

Comment:
I think that odd number of of cores is very unusual. Also should probably say something about "file content" rather than "output of command."

Reply:
The number of lines on output depends on the number of "online" files. I understand that cpu0 is always online, so it does not have the "online" file.
But I agree that the docstring can be misleading.

Reply:
I see. May be worth adding a note.